ctru-rs/src/error.rs

```diff
@@ -55,6 +55,8 @@ pub enum Error {
     Libc(String),
     ServiceAlreadyActive,
     OutputAlreadyRedirected,
+    /// The first member is the length of the buffer provided by the user, the second parameter is the size of the requested data (in bytes).
+    BufferTooShort(usize, usize),
```

Comment:
tiny nit, if you want to make these values self-describing this could be something like
```rs
BufferTooShort {
    provided: usize,
    wanted: usize,
}
```

Reply:
I don't believe this is correct Rust code (?). The struct would just be a member of the enum, and **_not_** a variant of the aforementioned. I've seen some postponed RFCs about this kind of use, but it's not yet available.

Reply:
The change @ian-h-chamberlain suggested is correct Rust code (unless I don't understand what you're referring to). [example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e8908f1f4219776f8c28aa99fafb9f56)

Reply:
Oops, yeah i worded it wrongly. IT IS correct rust code, it's just not what we want it to do, I think (it'd be a different type, not a variant of the enum).

Reply:
If I'm not wrong, both ``BufferTooShort(usize, usize)`` and ``BufferTooShort { provided: usize, wanted: usize }`` are variants of the enum. (Or maybe I still don't understand where is the problem 😅)

---
[From The Rust Reference](https://doc.rust-lang.org/reference/items/enumerations.html):

Enum constructors can have either named or unnamed fields:

```rust
enum Animal {
    Dog(String, f64),
    Cat { name: String, weight: f64 },
}

let mut a: Animal = Animal::Dog("Cocoa".to_string(), 37.2);
a = Animal::Cat { name: "Spotty".to_string(), weight: 2.7 };
```
In this example, Cat is a _struct-like **enum variant**_, whereas Dog is simply called an enum variant.

Reply:
You are absolutely correct, I had just misread a "struct" before the variant, which is definitely not there. Maybe I'm just tired 😅

Reply:
Catching up on some PRs, is there a reason this change wasn't incorporated? I see the `Debug` implementation pretends like it has named fields.

Reply:
It was implemented, but it was directly pushed to main after the PR was merged (unless I don't remember correctly)

Reply:
Yeah it’s in `master`. I don’t really remember why it doesn’t show here. It’s possible I pushed it afterwards.